src/test/scala/org/zalando/nakadi/client/KlientSpec.scala

```diff
@@ -9,53 +9,24 @@ import com.fasterxml.jackson.module.scala.DefaultScalaModule
 import com.google.common.collect.Iterators
 import com.typesafe.scalalogging.LazyLogging
 import io.undertow.util.{HeaderValues, HttpString, Headers}
-import org.scalatest.{BeforeAndAfterEach, Matchers, WordSpec}
+import org.scalatest._
+import org.scalatest.concurrent.PatienceConfiguration.{Timeout => ScalaTestTimeout}
+import org.scalatest.concurrent.ScalaFutures
 import org.zalando.nakadi.client.actor.PartitionReceiver
 import org.zalando.nakadi.client.utils.NakadiTestService
 import org.zalando.nakadi.client.utils.NakadiTestService.Builder
+import org.zalando.nakadi.client.utils.{Request => NakadiRequest}
 
 import scala.concurrent.Await
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
 
-class TestListener extends  Listener {
-  var receivedEvents = new AtomicReference[List[Event]](List[Event]())
-  var onConnectionClosed = 0
-  var onConnectionOpened = 0
-  var onConnectionFailed = 0
-
-  override def id = "test"
-
-  override def onReceive(topic: String, partition: String, cursor: Cursor, event: Event): Unit =  {
-    println(s"WAS CALLED [topic=$topic, partition=$partition, event=$event]" )
-
-    var old = List[Event]()
-    do {
-      old = receivedEvents.get()
-    }
-    while(! receivedEvents.compareAndSet(old, old ++ List(event)))
-  }
-  override def onConnectionClosed(topic: String, partition: String, lastCursor: Option[Cursor]): Unit = onConnectionClosed += 1
-  override def onConnectionOpened(topic: String, partition: String): Unit = onConnectionOpened += 1
-  override def onConnectionFailed(topic: String, partition: String, status: Int, error: String): Unit = onConnectionFailed += 1
-}
-
-
-class KlientSpec extends WordSpec with Matchers with BeforeAndAfterEach with LazyLogging {
+class KlientSpec extends WordSpec with Matchers with BeforeAndAfterEach with LazyLogging with ScalaFutures {
```

Comment:
oops, the ScalaFutures sneaked in here. The code's not using it, yet. It provides an alternative way to do what `Await.result` does. Will make a separate PR on that.